.clang-tidy

```diff
@@ -0,0 +1 @@
+Checks: clang-diagnostic-*,clang-analyzer-*,modernize-*,cppcoreguidelines-*,bugprone-*
```

Comment:
Seems a bit aggressive to enable everything right off the bat. Would recommend to enable the ones that are working and then enable "the desired checks" as code is fixed.

Reply:
I should have made it more clear from the start that I don't plan to enable all of them. I wanted to see what pops out and then someone else can make a decision about what to disable and what to keep. Trailing return types is one of the first I usually disable as well.